Allow writing out custom annotation fields to cif files

Croydon-Brixton opened the pull request:
feat: allow writing out custom annotation to cif file

padix-key commented:
Thanks for implementing the feature the feature! I would propose a slight change: Instead of a boolean flag to automatically include all custom annotations, I would propose an `extra_fields` parameter, where the included annotations are named, because

- it is more consistent with the `extra_fields` parameter in `get_structure()` and
- *Explicit is better than implicit* according to the Zen of Python.

Croydon-Brixton commented:
Great points, thank you for the feedback @padix-key -- I'll address those and update the PR

Croydon-Brixton commented:
✅  Alright @padix-key  - I've changed the API to use the `extra_fields` parameter as you suggested and added a mini test for the reading & writing of custom annotation.

padix-key commented on tests/structure/test_pdbx.py:
```diff
@@ -753,3 +753,29 @@ def _clear_encoding(category):
+    assert "my_custom_annotation" in atoms.get_annotation_categories()
+    assert np.all(
+        atoms.get_annotation("my_custom_annotation").astype(int)
+        == np.arange(atoms.array_length())
```
A small nitpick: I think it would be better to put `np.arange(atoms.array_length())` into an extra variable, to make clear to the reader, that this is the same as the annotation that was used in `set_annotation()`

Croydon-Brixton replied:
✅  Done

padix-key reviewed (commented):
This looks good, I would only propose a small change, to make the test more clear.

padix-key commented:
I see, that you are not in `CONTRIB.rst`, yet. You are welcome to add yourself if you like.

Croydon-Brixton commented:
Thank you -- it'd be an honour! 
I've added myself & implemented your suggested change & rebased. Should be good to merge from my side, after CI passes -- you're welcome to squash the commits ofc
